### PR TITLE
Improve performance of asset list & add auto lock settings

### PR DESCRIPTION
**Pull request by nop33**
Related issues:
- https://github.com/alephium/alephium-frontend/issues/19
- https://github.com/alephium/alephium-frontend/issues/397
- https://github.com/alephium/alephium-frontend/issues/523
- #127

### CODE REVIEW

File: apps/mobile-wallet/src/components/layout/FlashListScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { FlashList, FlashListProps } from '@shopify/flash-list'
```
Comment (mvaivre) [inline apps/mobile-wallet/src/components/layout/FlashListScreen.tsx:19]: FlashList, yeah!

File: apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  ])
+
+  const clearBackgroundTimer = () => {
+    if (!lockTimer) {
+      return
+    }
+    BackgroundTimer.clearTimeout(lockTimer)
+    lockTimer = undefined
+  }
+}
```
Comment (mvaivre) [inline apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx:112]: Nit: could be a tiny bit shorter

```ts
const clearBackgroundTimer = () => {
  if (lockTimer) {
    BackgroundTimer.clearTimeout(lockTimer)
    lockTimer = undefined
  }
}
```

File: apps/mobile-wallet/src/features/auto-lock/utils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const autoLockSeconds = [5, 15, 30, 60]
+
+export const autoLockSecondsOptions = [
+  {
+    label: 'Fast',
```
Comment (mvaivre) [inline apps/mobile-wallet/src/features/auto-lock/utils.ts:23]: Wouldn't 'Immediately' be more clear?

**Review by mvaivre — COMMENTED** [review]
Code looks good, just a few comments! Need to test.

File: apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx (hunk 2 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx:112]: true. I just copy+pasta from Metamask XD

File: apps/mobile-wallet/src/features/auto-lock/utils.ts (hunk 3 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/auto-lock/utils.ts:23]: Yes, and that's what Metamask has as well. But I didn't like it so much in the UI. I kinda liked "Fast" more. Not a strong opinion. WDYT?

File: apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx (hunk 2 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/auto-lock/useAutoLock.tsx]: Fixed.

File: apps/mobile-wallet/src/components/layout/FlashListScreen.tsx (hunk 1 above)
Comment (nop33) [reply apps/mobile-wallet/src/components/layout/FlashListScreen.tsx:19]: ⚡

File: apps/mobile-wallet/src/features/auto-lock/utils.ts (hunk 3 above)
Comment (mvaivre) [reply apps/mobile-wallet/src/features/auto-lock/utils.ts:23]: Ok

**Comment by mvaivre** [comment]
This runs smoothly! A couple of visual things I noticed:

| Pic | Observation |
|-------|------------------|
|  | The page title and intro isn't aligned with the content anymore. |
|  | This is not strictly linked to this PR, but when importing a wallet with QR Code, we lack the proper header title and intro (explaining this is the password used to lock the wallet on desktop). |

**Pull request merged**